examples/synthetics_results_example_test.go

```diff
@@ -0,0 +1,422 @@
+}
+
+func formatTestResultsSlice(trs []synthetics.TestResults) string {
+	var s []string
+	for _, tr := range trs {
+		s = append(s, formatTestResults(tr))
+	}
+	return fmt.Sprintf("{\n%v\n}", strings.Join(s, ", "))
+}
+
+func formatTestResults(tr synthetics.TestResults) string {
+	return fmt.Sprintf(
+		"{\n  test_id=%v\n  time=%v\n  health=%v\n  len(agents)=%v\n  agents=%v\n}",
+		tr.TestID, tr.Time, tr.Health, len(tr.Agents), formatAgentsResults(tr.Agents),
+	)
+}
+
+func formatAgentsResults(ars []synthetics.AgentResults) string {
+	var s []string
+	for _, ar := range ars {
+		s = append(s, fmt.Sprintf("{agent_id=%v health=%v len(tasks)=%v}", ar.AgentID, ar.Health, len(ar.Tasks)))
+	}
+	return fmt.Sprintf("{%v}", strings.Join(s, ", "))
+}
+
+func demonstrateSyntheticsNetworkMeshTestResults() error {
+	ctx := context.Background()
+	client, err := NewClient()
+	if err != nil {
+		return err
+	}
+
+	testID, err := pickNetworkMeshTestID(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	trs, err := getLastTenTestResults(ctx, client, testID)
+	if err != nil {
+		return err
+	}
+	if len(trs) == 0 {
+		fmt.Println("No mesh test results received - exiting example")
+		return nil
+	}
+
+	getAllAgentsResp, err := client.Synthetics.GetAllAgents(ctx)
+	if err != nil {
+		return fmt.Errorf("client.Synthetics.GetAllAgents: %w", err)
+	}
+
+	m, err := newMetricsMatrix(trs, getAllAgentsResp.Agents)
+	if err != nil {
+		return fmt.Errorf("new metrics matrix: %w", err)
+	}
+
+	err = printPingLatencyMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping latency matrix: %w", err)
+	}
+
+	err = printPingJitterMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping jitter matrix: %w", err)
+	}
+
+	err = printPingPacketLossMatrix(m)
+	if err != nil {
+		return fmt.Errorf("print ping packet loss matrix: %w", err)
+	}
+
+	return nil
+}
+
+func getLastTenTestResults(ctx context.Context, c *kentikapi.Client, testID string) ([]synthetics.TestResults, error) {
+	trs, err := c.Synthetics.GetResultsForTests(ctx, synthetics.GetResultsForTestsRequest{
+		TestIDs:   []models.ID{testID},
+		StartTime: time.Now().Add(-time.Hour * 240), // last 10 days
+		EndTime:   time.Now(),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("GetResultsForTests: %w", err)
+	}
+
+	if len(trs) == 0 {
+		return nil, nil
+	}
+
+	fmt.Println("Number of test results:", len(trs))
+	// latest test trs are returned first in the array
+	return trs[0:min(10, len(trs))], nil
+}
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+// metricsMatrix holds mesh test results for ping task.
+type metricsMatrix struct {
+	// agents hold agents data in the same order as cells.
+	agents []synthetics.Agent
+	// cells hold all "fromAgent" -> "toAgent" connection metrics.
+	cells map[string]map[string]*synthetics.PingResults
+}
+
+func newMetricsMatrix(trs []synthetics.TestResults, allAgents []synthetics.Agent) (metricsMatrix, error) {
+	agents, err := prepareAgents(trs[0], allAgents)
```

Comment:
Note that different time samples may have different number of agents reporting (e.g. due to configuration changes, agents being down or slowness in data collection). It is probably better to get the list of agents from the configuration of the test and ignore data from any other agents, leave blanks if results are not available.

Reply:
I noticed that the example stopped working because of it. I'll fix it tomorrow.

Reply:
Done: c7762d1
Unfortunately it doesn't work when test contains agents with dual IP - test results contain IP v4 and test configuration contain IP v6. So printed table has no data for connections to agents with IP v6.
It works well with IPv4 agents.